### PR TITLE
Class list revise

### PR DESCRIPTION
**Pull request by shamedone**
Updated Course, Student, AdvisingServlet, and HTMLFormatter
to allow for failed and repeated classes to show in student
class list. Failed classes that have not been retaken show in
red, and repeated classes are displayed in line separated by
"/".

### CODE REVIEW

**Comment by shamedone** [comment]
Hi Professor Puder, 

This is the next stage of changes I have for the advising project. This set allows for repeat and failed classes to show up in the course list for each student. Please let me know of any changes. Thanks!

File: src/main/java/edu/sfsu/db/CampusTestDB.java [hunk 1]
```diff
@@ -1,13 +1,28 @@
...
+import java.sql.SQLException;
+import java.util.ArrayList;
 import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
 
 public class CampusTestDB extends DB implements CampusDB {
 
+    final static protected String DB_NAME = "ADVISING";
```
Comment (apuder) [inline src/main/java/edu/sfsu/db/CampusTestDB.java]: I am a little bit confused. Those changes you have here were already in the previous PR. Can you please check and remove changes that were already merged? Thx.

**Review by apuder — CHANGES_REQUESTED** [review]

File: src/main/java/edu/sfsu/db/CampusTestDB.java (hunk 1 above)
Comment (shamedone) [reply src/main/java/edu/sfsu/db/CampusTestDB.java]: Ah I think I just forgot to push the merge with the previous master. I pushed the merge and the repeated section of the campusTestDb disappeared.

Comment (apuder) [reply src/main/java/edu/sfsu/db/CampusTestDB.java]: There are two commits now. Please rebase to master, squash into one commit, then force-push. Thx.

Comment (shamedone) [reply src/main/java/edu/sfsu/db/CampusTestDB.java]: Ok Should now be one commit and only the new stuff.

**Comment by shamedone** [comment]
Hi Arno, 

I followed the suggestions you emailed me about handling the student grade data. I have updated this pull request with the changes. They are a little different than what you suggested, namely that in the event a course has not been passed, the requirement class still returns an instance of the class with the grade sequence string and last semester attempted for that course. If this is not the desired behavior I will change it accordingly. 

Other than that, I think it is following what you have suggested, if I am an off base please let me know and we will discuss and I will make the changes accordingly. Thanks!

 -Tom

File: src/main/java/edu/sfsu/db/Student.java [hunk 2]
```diff
@@ -30,35 +31,40 @@ public Student(String id) {
...
+    private Requirement requirementFor(String courseName) {
+        Requirement req = new Requirement(courseName);
         for (Course course : courses) {
             if (course.courseName.equals(courseName)) {
                 if (course.grade.equals("")) {
                     // Course taken this semester
-                    wip = course;
-                    continue;
+                    //wip = course;
+                    req.addGrade("In Progres"); //mark class as in progress
```
Comment (apuder) [inline src/main/java/edu/sfsu/db/Student.java]: Use "IP" instead of "In Progress"

File: src/main/java/edu/sfsu/db/Requirement.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
+package edu.sfsu.db;
```
Comment (apuder) [inline src/main/java/edu/sfsu/db/Requirement.java:1]: Can you please reformat this file with IntelliJ's default format style?

File: src/main/java/edu/sfsu/HtmlFormatter.java [hunk 4]
```diff
@@ -120,6 +129,8 @@ public static String generateHtml(Student student) {
         generateClassList(fragment, student, "Core", CourseRequirements.core, true, true);
         generateClassList(fragment, student, "Electives", CourseRequirements.electives, false, true);
         generateClassList(fragment, student, "Transfers", transfers, true, false);
+
```
Comment (apuder) [inline src/main/java/edu/sfsu/HtmlFormatter.java]: Remove those two empty lines.

File: src/main/java/edu/sfsu/HtmlFormatter.java [hunk 5]
```diff
@@ -71,25 +72,33 @@ private static void generateClassList(HTMLSniplet fragment, Student student, Str
                         details.p("transfer", "");
                         details.p("semester", "");
                         details.p("grade", "");
+                        details.p("fail_flag", "passed");
+
```
Comment (apuder) [inline src/main/java/edu/sfsu/HtmlFormatter.java]: Remove empty line.

**Review by apuder — CHANGES_REQUESTED** [review]
Some formatting change requests. Please do the usual updates.

**Comment by shamedone** [comment]
Ok, formatting updated based on your comments. Thanks!

**Pull request merged**